File: examnation/lib/models/lesson.dart
class Lesson {
  final int id;
  final String title;
  final String subject;
  final String level;
  final String content;

  Lesson({
    required this.id,
    required this.title,
    required this.subject,
    required this.level,
    required this.content,
  });

  factory Lesson.fromJson(Map<String, dynamic> json) => Lesson(
    id: json['id'] is int ? json['id'] : int.parse(json['id'].toString()),
    title: json['title'] ?? '',
    subject: json['subject'] ?? '',
    level: json['level'] ?? '',
    content: json['content'] ?? '',
  );
}

File: backend/src/main/java/com/examnation/backend/model/Lesson.java
package com.examnation.backend.model;

import lombok.Data;
import jakarta.persistence.*;
import java.util.List;

@Entity
@Table(name = "lessons")
@Data
public class Lesson {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(nullable = false)
    private String subject;

    @Column(nullable = false)
    private String level;

    @OneToMany(mappedBy = "lesson", cascade = CascadeType.ALL)
    private List<FlashCard> flashCards;

    @OneToMany(mappedBy = "lesson", cascade = CascadeType.ALL)
    private List<McqQuiz> mcqQuizzes;
}